staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go

```diff
@@ -193,6 +194,18 @@ func (c *namedCreaterAdapter) Create(ctx context.Context, name string, obj runti
 	return c.Creater.Create(ctx, obj, createValidatingAdmission, options)
 }
 
+// manager is assumed to be already a valid value, we need to make
+// userAgent into a valid value too.
+func managerOrUserAgent(manager, userAgent string) string {
+	if manager != "" {
+		return manager
+	}
+	return prefixFromUserAgent(userAgent)
+}
+
+// prefixFromUserAgent takes the characters preceding the first /, quote
+// unprintable character and then trim what's beyond the
+// FieldManagerMaxLength limit.
 func prefixFromUserAgent(u string) string {
-	return strings.Split(u, "/")[0]
+	return strconv.Quote(strings.Split(u, "/")[0])[:validation.FieldManagerMaxLength]
```

Comment:
Did you test this on a short string?

Reply:
The one thing I didn't test :-//

Reply:
OK Fixed, and the hamburgers are not cut either.